### PR TITLE
feat(output): support compact stdout/json output

### LINKED ISSUE #107: Compact STDOUT/JSON Output
## Feature Request

Add `--compact` flag to CLI to enable abbreviated stdout.

## Is your feature request related to a problem? Please describe.

The default outputs product such large output (300K+ lines) which can be painful to review.

## Describe the solution you'd like

Add `--compact` flag to CLI to display the most relevant information used to ignore false flags.

_STDOUT_

`[{IssueType} ({Signature})] {file_path}@L{matched_line_no}: {matched_line}`

```text
[High Entropy (49d8990db16295778fdb98560a4b9149ca9161c88423b6dc5cc4ffce74521483)] src/file1.js@L1: // flow-typed signature: dc381ee55406f66b7272c6343db0834b
```

_JSON_

```json
{
  "found_issues": [
    {
      "file_path": "src/file1.js",
      "matched_string": "dc381ee55406f66b7272c6343db0834b",
      "matched_line": "// flow-typed signature: dc381ee55406f66b7272c6343db0834b",
      "matched_line_no": 1,
      "issue_type": "High Entropy",
      "issue_detail": ""
    }
  ]
}
```

## Describe alternatives you've considered

The current json/stdout.

### PR DESCRIPTION
**Pull request by dclayton-godaddy**
Add support for compact stdout and JSON output. Adds `--compact/--no-compact` to CLI defaulting to current behavior/no-compact.

### CODE REVIEW

**Review by sjacoby-godaddy — APPROVED** [review]
lgtm

File: tartufo/scanner.py [hunk 1]
```diff
@@ -49,13 +49,33 @@ def __init__(
...
 
-    def as_dict(self) -> Dict[str, Optional[str]]:
+    def _as_compact_dict(self) -> Dict[str, Optional[str]]:
+        """Return a compact dictionary representation of an issue.
+
+        This is primarily meant to aid in JSON serialization with fewer fields.
+
+        :return: A compact JSON serializable dictionary representation of this issue
+        """
+        return {
```
Comment (jmink-godaddy) [inline tartufo/scanner.py]: This seems like code duplication to me. If the compact dict will always be a subset of the non-compact, I would prefer that the same code populates the fields in common.

**Comment by tarkatronic** [comment]
Given how many changes we've had in `main` since this was submitted, I went ahead and rebased it, and fixed all the conflicts/errors. @jmink-godaddy I can go ahead and take care of your concern as well and then I think we're good to go on getting this merged!

File: tartufo/scanner.py (hunk 1 above)
Comment (tarkatronic) [reply tartufo/scanner.py]: @jmink-godaddy Pretty sure I got this resolved for ya. :)

**Review by jgowdy — APPROVED** [review]
Looks good.  Thanks for getting this across the finish line @tarkatronic

File: tartufo/util.py [hunk 2]
```diff
@@ -66,9 +66,18 @@ def echo_result(
...
+            "found_issues": [
+                issue.as_dict(compact=options.compact) for issue in scanner.issues
+            ],
         }
+
         click.echo(json.dumps(output))
+    elif options.compact:
+        for issue in scanner.issues:
+            click.echo(
+                f"[{issue.issue_type.value}] {issue.chunk.file_path}: {issue.matched_string} "
```
Comment (sjacoby-godaddy) [inline tartufo/util.py:78]: not sure if this trailing space is intentional

**Review by sjacoby-godaddy — APPROVED** [review]
looks good! one non-blocking comment

File: tartufo/util.py (hunk 2 above)
Comment (tarkatronic) [reply tartufo/util.py:78]: Yup! This is using implicit string joining to the next line, so these lines end up looking like (extracting from the tests):
> [Regular Expression Match] /what/bar: bar (fa692eebc3d60e67a9f22b4b877d5939cb2ec96c0c26c7e5168b3b8b660c573c, Meets the bar)

**Pull request merged**